kolibri/content/models.py

```diff
@@ -0,0 +1,333 @@
+"""
+This is one of the Kolibri core components, the abstract layer of all contents.
+To access it, please use the public APIs in api.py
+
+The ONLY public object is ContentMetadata
+"""
+from __future__ import print_function
+
+import hashlib
+import os
+from uuid import uuid4
+
+from django.conf import settings
+from django.contrib import admin
+from django.core.files.storage import FileSystemStorage
+from django.db import IntegrityError, connections, models
+from django.db.utils import ConnectionDoesNotExist
+from mptt.models import MPTTModel, TreeForeignKey
+
+
+def content_copy_name(instance, filename):
+    """
+    Create a name spaced file path from the File obejct's checksum property.
+    This path will be used to store the content copy
+
+    :param instance: File (content File model)
+    :param filename: str
+    :return: str
+    """
+    h = instance.checksum
+    basename, ext = os.path.splitext(filename)
+    return os.path.join(settings.CONTENT_COPY_DIR, h[0:1], h[1:2], h + ext.lower())
+
+class ContentCopyStorage(FileSystemStorage):
+    """
+    Overrider FileSystemStorage's default save method to ignore duplicated file.
+    """
+    def get_available_name(self, name):
+        return name
+
+    def _save(self, name, content):
+        if self.exists(name):
+            # if the file exists, do not call the superclasses _save method
+            print('file "', name, '" already exists!')
```

Comment:
Please use ```logging.warn``` for this, not print.

I assume the point of this is that saving a file that already exists just shortcuts the actual saving?